File: java-code/src/main/java/refactoring/conditional_logic/replace_conditional_with_polymorphism/after/AfricanSwallow.java
package refactoring.conditional_logic.replace_conditional_with_polymorphism.after;

public class AfricanSwallow extends Bird {

  public AfricanSwallow(Bird bird) {
    super(bird.getName(), bird.getType(), bird.getNumberOfCoconuts(), bird.getVoltage(), bird.isNailed());
  }
  @Override
  public String plumage() {
    return (this.getNumberOfCoconuts() > 2) ? "tired" : "average";
  }

  @Override
  public int airSpeedVelocity() {
    return 40 * this.getNumberOfCoconuts();
  }
}

File: go-code/conditional_logic/replace_conditional_with_polymorphism/example1/after/AfricanSwallow.go
package after

type AfricanSwallow struct {
	BirdAttributes
}

func (a AfricanSwallow) GetBirdAttributes() BirdAttributes {
	return a.BirdAttributes
}

func (a AfricanSwallow) Plumage() string {
	if a.NumberOfCoconuts > 2 {
		return "tired"
	}
	return "average"
}

func (a AfricanSwallow) AirSpeedVelocity() int {
	return 40 * a.NumberOfCoconuts
}
